addons/portal/controllers/portal_thread.py

```diff
@@ -67,11 +67,16 @@ def portal_message_fetch(self, thread_model, thread_id, fetch_params=None, **kw)
         # Check access
         Message = request.env['mail.message']
         if kw.get('token'):
-            access_as_sudo = ThreadController._get_thread_with_access(
+            thread = ThreadController._get_thread_with_access(
                 thread_model, thread_id, token=kw.get("token"),
             )
-            if not access_as_sudo:  # if token is not correct, raise Forbidden
+            if not thread:  # if token is not correct, raise Forbidden
                 raise Forbidden()
+            if request.env.user._is_public():
```

Comment:
Technically we could in theory have a portal partner for a different partner than current user, even as non-public user?

Reply:
I don't think so. If the user is not public, the user's partner has priority. If there isn't one (in the case of a public user), the partnert_id of the document (token case) or the `pid` of the document (hash and pid case) would be authenticated as the portal partner.

Reply:
The logged in user should take priority when we need to _assign_ a user on a field (new message author, reaction ...).

But when determining who is the author of a message, both could be taken into account.
If we don't check it, it would mean it's possible to edit a portal partner message by logging out (and becoming portal parter) and not by staying logged in as another user (as it takes priority).
It's rather confusing and counter-productive to gain more access to something by logging out.